operators/pkg/tenant-controller/tenant_controller.go

```diff
@@ -144,27 +154,87 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		retrigErr = err
 	}
 
+
+
+	keepNsOpen := true; // keepNsOpen defines if we should close the personal namespace based on the last login date
+	                    // must be initialized for later use
+
+	// If the personalNamespace is not created and the lastLogin is blank, then this is the first time
+	// the tenant has logged in; update the lastLogin date to current time
+	if tn.Status.PersonalNamespace.Created==false && tn.Spec.LastLogin=="" {
+
+	  tn.Spec.LastLogin = time.Now().Format(time.RFC3339)
+      klog.Infof("First login of %s: setting lastLogin value of %s in tenant spec.", tn.Name, tn.Spec.LastLogin)
+
+	} else { // Otherwise, this is not the first time the user has logged in
```

Comment:
Hmm, I don't exactly understand, then should this code be removed? Or should we keep the code here and assume that the user's first login would be when the tenant is created?

Or, as a third solution I would propose, what if we make the lastLogin upon tenant creation `0` so that way it does not force the namespace creation?

Reply:
Yes, I was also proposing your third solution (the default if not set is already 0)

Reply:
Resolved in [92b5634](https://github.com/netgroup-polito/CrownLabs/pull/821/commits/92b563429b934c21661528ab29a2ec5894148cb2)